internal/resources/resources_test.go

```diff
@@ -0,0 +1,138 @@
+package resources
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	networkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+var sc *runtime.Scheme
+
+var _ = Describe("Resources", func() {
+	sc = runtime.NewScheme()
+	Expect(networkingv1alpha3.AddToScheme(sc)).To(Succeed())
+
+	DescribeTable("FindUserCreatedIstioResourcesDescribe", func(ctx context.Context, logger logr.Logger, client client.Client, configuration resourceFinderConfiguration, want []Resource, wantErr bool) {
+		i := &IstioResourcesFinder{
+			ctx:           ctx,
+			logger:        logger,
+			client:        client,
+			configuration: configuration,
+		}
+		got, err := i.FindUserCreatedIstioResources()
+		Expect(err != nil).To(Equal(wantErr))
+		Expect(got).To(BeEquivalentTo(want))
+	},
+		Entry("Should get nothing if there are only default istio resources present", context.TODO(),
```

Comment:
should we add also test when we find a customer created resource?

Reply:
Yes